### PR TITLE
adds test for the create, Update & delete REST API

### PR DESCRIPTION
**Pull request by eranwitkon**
This PR add the required tests for REST API CRUD operation.
This replace previous PR's which where closed due to merge issues.

### CODE REVIEW

File: zeppelin-server/src/test/java/org/apache/zeppelin/rest/AbstractTestRestApi.java [hunk 1]
```diff
@@ -322,6 +347,13 @@ protected void describeMismatchSafely(String item, Description description) {
     };
   }
 
+  //Create new Setting and return Setting ID
+  protected String createTempSetting(String tempName) throws IOException {
+
+    InterpreterGroup interpreterGroup =  ZeppelinServer.notebook.getInterpreterFactory().add(tempName,"newGroup",new InterpreterOption(false),new Properties());
```
Comment (bzz) [inline zeppelin-server/src/test/java/org/apache/zeppelin/rest/AbstractTestRestApi.java]: This line seems to be longer then 100 chars, which is discouraged by [Styleguide](https://github.com/apache/incubator-zeppelin/blob/master/CONTRIBUTING.md#code-convention)

**Comment by eranwitkon** [comment]
Fixed and ready for review.

**Comment by bzz** [comment]
Are you sure that things like 754c55e and  4f3c840  a849b4b d1d32b1 are intended to be part of this PR\contribution?

**Comment by eranwitkon** [comment]
This wasn't my intention,  I am only changing 2 files but this was probebly due to rebase on master. I will try and see how to fix it.
Sorry.

**Comment by bzz** [comment]
No problem, enthusiastic :+1: for tests!

Yes, interactive rebasing might be a good idea or it might be even simpler to re-create this PR from the latest master, cherry-picking the particular commits.

**Comment by eranwitkon** [comment]
Unneeded commits in the PR, closing and re-creating from master.

**Comment by eranwitkon** [comment]
Just closed the PR and opened a new one PR #286.
Hope it will pass CI and ready for merge.
Eran

On Mon, Sep 7, 2015 at 7:35 AM Alexander <notifications@github.com> wrote:

> No problem, enthusiastic [image: :+1:] for tests!
>
> Yes, interactive rebasing might be a good idea or it might be even simpler
> to re-create this PR from the latest master, cherry-picking the particular
> commits.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-zeppelin/pull/277#issuecomment-138184596>
> .
>
-- 
Eran | "You don't need eyes to see, you need vision" (Faithless)

**Pull request closed without merging**